player/lua/input.lua

```diff
@@ -0,0 +1,69 @@
+--[[
+This file is part of mpv.
+
+mpv is free software; you can redistribute it and/or
+modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation; either
+version 2.1 of the License, or (at your option) any later version.
+
+mpv is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+local utils = require "mp.utils"
+local input = {}
+
+function input.get(t)
+    local console_enabled = utils.shared_script_property_get("console-enabled")
+    if console_enabled == nil then
+        return nil, "The console script is not loaded."
+    end
+    if console_enabled == "1" then
+        return nil, "The console is already open."
+    end
+
+    mp.commandv("script-message-to", "console", "get-input", mp.get_script_name(),
+                t.prompt or "", t.id or "")
```

Comment:
No need for nil checks, your `get-input` handler already deals with nil parameters with `a ~= '' and a or b`. Same is probably also true on the javascript side, but I haven't tested that.

Reply:
`script-message` errors with `nil` arguments.

Reply:
Oh, I didn't know that and it's also not mentioned in the documentation (or I missed it).

Reply:
JSON could also be used for this script-message to make it easier to add new fields in the future.

Reply:
This was fine for a few arguments. I will switch to JSON to add the default text and cursor arguments.

Reply:
Done. Note that in Lua the JSON variables aren't inlined because `format_json` returns `nil` as the second argument, which makes `script-message` error.

Reply:
You can avoid that behaviour by wrapping the function call in brackets:

```lua
mp.commandv("script-message-to", "console", "get-input", mp.get_script_name(), (utils.format_json({
        prompt = t.prompt,
        default_text = t.default_text,
        cursor_position = t.cursor_position,
        id = t.id,
    })))
```

It strips all the additional return values.

Reply:
Yes, I compared how you do it in your script after writing that, but hopefully https://github.com/mpv-player/mpv/pull/13123 is merged and we can do this normally.

Reply:
Got rid of the variables.